tests/integration/test_spansv2.py

```diff
@@ -537,3 +538,148 @@ def test_spanv2_inbound_filters(
     ]
 
     assert mini_sentry.captured_events.empty()
+
+
+@pytest.mark.parametrize(
+    "rule_type,test_value,expected_scrubbed",
+    [
+        ("@ip", "127.0.0.1", "[ip]"),
+        ("@email", "test@example.com", "[email]"),
+        ("@creditcard", "4242424242424242", "[creditcard]"),
+        ("@iban", "DE89370400440532013000", "[iban]"),
+        ("@mac", "4a:00:04:10:9b:50", "*****************"),
+        (
+            "@uuid",
+            "ceee0822-ed8f-4622-b2a3-789e73e75cd1",
+            "************************************",
+        ),
+        ("@imei", "356938035643809", "[imei]"),
+        (
+            "@pemkey",
+            "-----BEGIN EC PRIVATE KEY-----\nMIHbAgEBBEFbLvIaAaez3q0u6BQYMHZ28B7iSdMPPaODUMGkdorl3ShgTbYmzqGL\n-----END EC PRIVATE KEY-----",
+            "-----BEGIN EC PRIVATE KEY-----\n[pemkey]\n-----END EC PRIVATE KEY-----",
+        ),
+        (
+            "@urlauth",
+            "https://username:password@example.com/",
+            "https://[auth]@example.com/",
+        ),
+        ("@usssn", "078-05-1120", "***********"),
+        ("@userpath", "/Users/john/Documents", "/Users/[user]/Documents"),
+        ("@password", "my_password_123", ""),
+        ("@bearer", "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9", "Bearer [token]"),
+    ],
+)
+def test_spanv2_with_string_pii_scrubbing(
+    mini_sentry,
+    relay,
+    rule_type,
+    test_value,
+    expected_scrubbed,
+):
+    project_id = 42
+    project_config = mini_sentry.add_full_project_config(project_id)
+    project_config["config"]["features"] = [
+        "organizations:standalone-span-ingestion",
+        "projects:span-v2-experimental-processing",
+    ]
+
+    project_config["config"]["piiConfig"]["applications"] = {"$string": [rule_type]}
+
+    relay = relay(mini_sentry, options=TEST_CONFIG)
+    ts = datetime.now(timezone.utc)
+
+    envelope = envelope_with_spans(
+        {
+            "start_timestamp": ts.timestamp(),
+            "end_timestamp": ts.timestamp() + 0.5,
+            "trace_id": "5b8efff798038103d269b633813fc60c",
+            "span_id": "eee19b7ec3c1b174",
+            "name": "Test span",
+            "is_remote": False,
+            "attributes": {
+                "test_pii": {"value": test_value, "type": "string"},
+            },
+        }
+    )
+
+    relay.send_envelope(project_id, envelope)
+
+    envelope = mini_sentry.captured_events.get()
+    item_payload = json.loads(envelope.items[0].payload.bytes.decode())
+    item = item_payload["items"][0]
+    attributes = item["attributes"]
+
+    assert "test_pii" in attributes
+    assert attributes["test_pii"]["value"] == expected_scrubbed
+    assert "_meta" in item
+    meta = item["_meta"]["attributes"]["test_pii"]["value"][""]
+    assert "rem" in meta
+
+    # Check that the rule type is mentioned in the metadata
+    rem_info = meta["rem"][0]
+    assert rule_type in rem_info[0]
```

Comment:
I know this is copy paste from logs, should eventually clean this up. Can just assert the payload instead of picking individual items.

Reply:
Done.